core/templates/domain/skill/SkillObjectFactory.ts

```diff
@@ -13,266 +13,292 @@
-      this._description = skill.getDescription();
-      this._misconceptions = skill.getMisconceptions();
-      this._rubrics = skill.getRubrics();
-      this._conceptCard = skill.getConceptCard();
-      this._languageCode = skill.getLanguageCode();
-      this._version = skill.getVersion();
-      this._nextMisconceptionId = skill.getNextMisconceptionId();
-      this._supersedingSkillId = skill.getSupersedingSkillId();
-      this._allQuestionsMerged = skill.getAllQuestionsMerged();
-      this._prerequisiteSkillIds = skill.getPrerequisiteSkillIds();
-    };
-
-    // TODO(ankita240796): Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    Skill['createFromBackendDict'] = function(skillBackendDict) {
-    /* eslint-enable dot-notation */
-      return new Skill(
-        skillBackendDict.id,
-        skillBackendDict.description,
-        generateMisconceptionsFromBackendDict(skillBackendDict.misconceptions),
-        generateRubricsFromBackendDict(skillBackendDict.rubrics),
-        ConceptCardObjectFactory.createFromBackendDict(
-          skillBackendDict.skill_contents),
-        skillBackendDict.language_code,
-        skillBackendDict.version,
-        skillBackendDict.next_misconception_id,
-        skillBackendDict.superseding_skill_id,
-        skillBackendDict.all_questions_merged,
-        skillBackendDict.prerequisite_skill_ids);
-    };
-
-
-    // Create an interstitial skill that would be displayed in the editor until
-    // the actual skill is fetched from the backend.
-    // TODO(ankita240796): Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    Skill['createInterstitialSkill'] = function() {
-    /* eslint-enable dot-notation */
-      return new Skill(null, 'Skill description loading',
-        [], [], ConceptCardObjectFactory.createInterstitialConceptCard(), 'en',
-        1, 0, null, false, []);
-    };
-
-    var generateMisconceptionsFromBackendDict = function(
-        misconceptionsBackendDicts) {
-      return misconceptionsBackendDicts.map(function(
-          misconceptionsBackendDict) {
-        return MisconceptionObjectFactory.createFromBackendDict(
-          misconceptionsBackendDict);
-      });
-    };
-
-    var generateRubricsFromBackendDict = function(rubricBackendDicts) {
-      return rubricBackendDicts.map(function(rubricBackendDict) {
-        return RubricObjectFactory.createFromBackendDict(rubricBackendDict);
-      });
-    };
+export interface ISkillBackendDict {
+  'all_questions_merged': boolean;
+  'description': string;
+  'id': string;
+  'language_code': string;
+  'misconceptions': IMisconceptionBackendDict[];
+  'next_misconception_id': number;
+  'prerequisite_skill_ids': string[];
+  'rubrics': IRubricBackendDict[];
+  'skill_contents': IConceptCardBackendDict;
+  'superseding_skill_id': string;
+  'version': number;
+}
+
+import { ConceptCardObjectFactory, ConceptCard,
```

Comment:
Please sort the imports. 

All angular dependencies should be at the top, a line break and then oppia dependencies. All in alphabetical order.

Reply:
sorted the import 
thanks